pkg/target/host.go

```diff
@@ -90,63 +90,61 @@ func (h *Host) Stream(wg *sync.WaitGroup) {
 	}()
 
 	for r := range h.Ch {
-		// retry until successful
-		for {
-			const maxReconnectPeriodMs = 5000
-			const reconnectDeltaMs = 10
-			for reconnectWait := 0; h.Conn == nil; {
-				time.Sleep(time.Duration(reconnectWait) * time.Millisecond)
-				if reconnectWait < maxReconnectPeriodMs {
-					reconnectWait = reconnectWait*2 + reconnectDeltaMs
-				}
-				if reconnectWait >= maxReconnectPeriodMs {
-					reconnectWait = maxReconnectPeriodMs
-				}
+		h.tryToSend(r)
+	}
+}
 
-				h.Connect()
-			}
+func (h *Host) tryToSend(r *rec.Rec) {
+	h.CWm.Lock()
+	defer h.CWm.Unlock()
 
-			err := h.Conn.SetWriteDeadline(time.Now().Add(
-				time.Duration(h.SendTimeoutSec) * time.Second))
-			if err != nil {
-				h.Lg.Warn("error setting write deadline. Renewing the connection.",
-					zap.Error(err))
+	// retry until successful
+	for {
+		const maxReconnectPeriodMs = 5000
+		const reconnectDeltaMs = 10
+		for reconnectWait := 0; h.Conn == nil; {
+			time.Sleep(time.Duration(reconnectWait) * time.Millisecond)
+			if reconnectWait < maxReconnectPeriodMs {
+				reconnectWait = reconnectWait*2 + reconnectDeltaMs
+			}
+			if reconnectWait >= maxReconnectPeriodMs {
+				reconnectWait = maxReconnectPeriodMs
 			}
 
-			// this may loose one record on disconnect
-			_, err = h.Write([]byte(*r.Serialize()))
+			h.Connect()
+		}
 
-			if err == nil {
-				h.outRecs.Inc()
-				h.processingDuration.Observe(time.Since(r.Received).Seconds())
-				break
-			}
+		err := h.Conn.SetWriteDeadline(time.Now().Add(
+			time.Duration(h.SendTimeoutSec) * time.Second))
+		if err != nil {
+			h.Lg.Warn("error setting write deadline. Renewing the connection.", zap.Error(err))
```

Comment:
Where do we renew the connection?
It seems that we just log.

Reply:
Firstly, this is the same code as before. It is unchanged.

Do we need to renew the connection here? This signals that setting the deadline failed. Does that mean we need to renew?

Reply:
I never intended to renew the connection here.

Reply:
This is a different change than this one. I agree that it may make sense but I would keep this separate from the current PR.

Reply:
I got a bit confused by the difference between the log message and the action.
We log that we will renew connection, but we don't do anything.

We should probably not try to write to the buffer if the connection is broken as this data would 100% get lost. 

Feel free to open issue to fix later, update log line, leave it as it is.

Reply:
Yes, the message is wrong.

Reply:
Fixed.